### PR TITLE
SW-2229 Database schema for planting site deliveries

### PR DESCRIPTION
**Pull request by sgrimm**
Create the tables to hold information about deliveries to planting sites. Nothing
accesses these tables yet.

In addition to the new tables, add a `planting_site_id` column to the `plots`
table to enable us to have a foreign-key constraint that prevents a planting from
having a planting site ID that doesn't match its plot ID.

### CODE REVIEW

**Comment by sgrimm** [comment]
Current dependencies on/for this PR:
* main
  * **PR #810** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/810" target="_blank">Graphite</a>  👈
    * **PR #811** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/811" target="_blank">Graphite</a> 
      * **PR #812** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/812" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/810?utm_source=stack-comment).

File: src/main/resources/db/migration/V159__Deliveries.sql [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+                        WHERE p.planting_zone_id = pz.id)
+WHERE planting_site_id IS NULL;
+
+ALTER TABLE tracking.plots
+    ALTER COLUMN planting_site_id SET NOT NULL;
+ALTER TABLE tracking.plots
+    ADD FOREIGN KEY (planting_site_id, planting_zone_id)
+        REFERENCES tracking.planting_zones (planting_site_id, id);
+ALTER TABLE tracking.plots
+    ADD UNIQUE (planting_site_id, id);
```
Comment (charlie-tf) [inline src/main/resources/db/migration/V159__Deliveries.sql]: Does there need to be a `UNIQUE (planting_zone_id, id)` constraint as well?

Comment (charlie-tf) [reply src/main/resources/db/migration/V159__Deliveries.sql]: Or is that sorta transitively enforced by 
```sql
ALTER TABLE tracking.planting_zones
    ADD UNIQUE (planting_site_id, id);
```
?

**Review by charlie-tf — APPROVED** [review]
Looks good! Just a minor question as I try to understand sql better

File: src/main/resources/db/migration/V159__Deliveries.sql (hunk 1 above)
Comment (sgrimm) [reply src/main/resources/db/migration/V159__Deliveries.sql]: Updated the script to hopefully answer this question; let me know if it's still unclear.

Comment (charlie-tf) [reply src/main/resources/db/migration/V159__Deliveries.sql]: That does help. Thank you!

**Comment by sgrimm** [comment]
**Graphite Merge Job**

Current status: ✅ Merged

This pull request was successfully merged as part of a stack.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/810?utm_source=stack-comment).


Job Reference: 1mEbeKkUgxrPyjlNq3Ug

**Pull request merged**